### PR TITLE
Revert "[CI] Use GitHub Actions (#2884)"

### PR DESCRIPTION
**Pull request by tmspzz**
This reverts commit e4b4c2f12b82d795487bcbeda428477fabc8123d.

### CODE REVIEW

**Comment by tmspzz** [comment]
I have restored the branch https://github.com/Carthage/Carthage/pull/2884 the PR will come as soon as this one is reverted

File: IntegrationTests/update.bats [hunk 1]
```diff
@@ -4,8 +4,8 @@ setup() {
     cd $BATS_TMPDIR
     rm -rf UpdateTest
     mkdir UpdateTest && cd UpdateTest
-    echo 'github "antitypical/Result" == 5.0.0' > Cartfile
-    echo 'github "Quick/Nimble" == 8.0.4' > Cartfile.private
+    echo 'github "antitypical/Result" == 4.1.0' > Cartfile
+    echo 'github "Quick/Nimble" == 7.3.1' > Cartfile.private
```
Comment (jdhealy) [inline IntegrationTests/update.bats]: Think we want to keep these

Comment (jdhealy) [reply IntegrationTests/update.bats]: And others

**Comment by jdhealy** [comment]
✨ Thanks for picking this up Tommaso!

**Comment by tmspzz** [comment]
I have kept the changes in the bats files and removed the fixtures.

**Comment by jdhealy** [comment]
Small thing: think https://github.com/Carthage/Carthage/pull/2903/files#diff-72e101a354524c7abb68b2b4070ca49e should be reverted to as in https://github.com/Carthage/Carthage/commit/e4b4c2f12b82d795487bcbeda428477fabc8123d — not removed

**Comment by tmspzz** [comment]
@jdhealy  should be done

**Comment by tmspzz** [comment]
@ikesyo please merge if you feel it's correct

**Comment by jdhealy** [comment]
Does the following commit message state our purpose well enough?

`[CI] React to GitHub Action depreciation of Xcode 10.1 by re-instating Travis-CI (somewhat temporarily).`

`Eventual CI runs should cover at least [which Xcode versions are required of us by Homebrew — see `OS::Mac::Xcode.latest_version` for macOS 10.13 and above](https://github.com/Homebrew/brew/commits/aae9d39/Library/Homebrew/os/mac/xcode.rb).`

**Pull request merged**